### ISSUE TITLE
GDPR prevents subscribing to youtube URLs with @ names

### ISSUE THREAD

**Issue by computerfreak105** [gpodder/gpodder#1448]
[2022-12-23.log](https://github.com/gpodder/gpodder/files/10297038/2022-12-23.log)
Hello!
If I'm not mistaken it's possible to subscribe to youtube channels in GPodder GUI by entering youtube channels in several ways. It doesn't necessarily have to be a channel ID like
https://www.youtube.com/channel/UCIzJqMXr6rT7PMgidXBFmtw

So, since youtube has introduced handles almost every channel has got new urls like
https://www.youtube.com/@paperworld2676
or
https://www.youtube.com/@user-og8ss9cj2t

Until now I used this online Tool here to find the channel ID of new youtube channels:

https://commentpicker.com/youtube-channel-id.php	

But I thought it would be nice if we could subscribe directly without having to use this tool.
So I tried this with GPO on CLI, because I wanted to see how this would look when I export the subscription as OPML-file.
GPO crashes when entering urls this way. (see attached log).

I tried to make a subscription with the GPodder GUI, but it's not accessible enough on windows for blind people and I think it didn't work - at least there was no subscription in the OPML file I exported afterwards.

So this is a report because of the crashing and a question, if it's possible at all to subscribe with the new urls youtube offers.
Thanks.

**Comment by auouymous** [comment]
I think this worked when the feature was announced, are you using 3.11 or an older version? Did you subscribe with GPO from the command line or from within its command line interface? The former might not work on Windows because I think @ is a special character. If that is the case, try wrapping the URL in quotes or subscribe from within gpo.

I'll take a look in a few days after I fix my computer.

**Comment by computerfreak105** [comment]
I'm using the portable version build artifact version 1.0.932.
And I start GPO from the windows explorer by opening the shortcut GPO from the portable version's folder.

Usually I prepare an OPML file and put all my channels I want to subscribe in there. Afterwards I'm importing the OPML file into GPO by writing import filename.opml.

"A line for a subscription looks like:

        <outline title="Cactus ASMR, Eng" text="No description available" xmlUrl="https://www.youtube.com/feeds/videos.xml?channel_id=UCsMNy2GLFpnonnJJf2cnAQg" type="rss"/>

The reason I do that is because I can decide the title of channels myself. This becomes handy for me because youtube channels may alter their title or they may contain foreigen language and unicode characters which I don't like having in a folder name.

Thanks for having a look.

**Comment by auouymous** [comment]
I didn't notice it before, but you mentioned using 932, which as you now know is a broken PR. I have tested and the new URLs work in gtk and cli.

But while testing gpo, I noticed that gtk is being pulled into gpo, and will get this fixed ASAP.

**Comment by computerfreak105** [comment]
Awesome, have many thanks.

**Comment by auouymous** [comment]
> But while testing gpo, I noticed that gtk is being pulled into gpo, ...

733cdd216977d1e2bab4c9d586598cc861a602d9 fixes it in https://ci.appveyor.com/project/elelay/gpodder/builds/45778151

**Comment by computerfreak105** [comment]
I’m afraid GPO still crashes when I try to import such new urls.

Should it work now from CLI or is it only working when running GPO in this GTK? I probably never have done this so far.

 

I run it from the windows explorer like just running “d:\gpo portable 1.0.933\gpo”. Here I can import opml-files. And when I’m done I run

“d:\gpo portable 1.0.933\gpo”. Here I can import opml-files.

And when I’m done I run “d:\gpo portable 1.0.933\gpodder” and let it run all the time.

 

I just was able with OCR trying to import from the gpodder GUI. But it didn’t work either. It said “Some urls couldn’t be added”. At least it didn’t crash and close like GPO.

 

 

 

 

 

 

Von: auouymous ***@***.***> 
Gesendet: Montag, 26. Dezember 2022 22:05
An: gpodder/gpodder ***@***.***>
Cc: computerfreak105 ***@***.***>; Author ***@***.***>
Betreff: Re: [gpodder/gpodder] GPO CLI crashes when entering youtube channel which doesn't contain channel ID (Issue #1448)

 

But while testing gpo, I noticed that gtk is being pulled into gpo, ...

 <https://github.com/gpodder/gpodder/commit/733cdd216977d1e2bab4c9d586598cc861a602d9> 733cdd2 fixes it in https://ci.appveyor.com/project/elelay/gpodder/builds/45778151

**Comment by auouymous** [comment]
It has worked fine in both cli and gtk versions since the new URLs were introduced. Can you run `gpo -v`, subscribe to the URL, and then post the log file here? This will spam the console with a lot of log messages.

**Comment by computerfreak105** [comment]
So, I tried with this url here

 

***@***.***

 

I send the log and here below I just copied the contents of the CLI.

Thanks, hope it helps.

 

 

Microsoft Windows [Version 10.0.19044.2364]

(c) Microsoft Corporation. Alle Rechte vorbehalten.

 

C:\Users\Benjamin Blatter>D:

 

D:\>cd gpodder-3.11.0-portable 933

 

D:\gpodder-3.11.0-portable 933>cd data

 

D:\gpodder-3.11.0-portable 933\data>cd bin

 

D:\gpodder-3.11.0-portable 933\data\bin>gpo -v

Calling SetDllDirectoryW(D:\gpodder-3.11.0-portable 933\data\bin)

Successfully SetDllDirectoryW

D: windows-portable build; forcing home to config directory D:\gpodder-3.11.0-portable 933\config

1672124306.168332 [gpodder.log] DEBUG: ==== gPodder starts up (ui=) ===

1672124306.313445 [gpodder.registry] DEBUG: Registering download_url resolver: <function youtube_real_download_url at 0x03e36fa0>

1672124306.318474 [gpodder.registry] DEBUG: Registering download_url resolver: <function vimeo_real_download_url at 0x03e307c0>

1672124306.368575 [gpodder.sync] INFO: iPod sync not available

1672124306.373578 [gpodder.sync] INFO: eyeD3 MP3 not available

1672124306.378590 [gpodder.extensions] DEBUG: Found extension "audio_converter" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\audio_converter.py

1672124306.378590 [gpodder.extensions] DEBUG: Found extension "command_on_download" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\command_on_download.py

1672124306.388699 [gpodder.extensions] DEBUG: Found extension "concatenate_videos" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\concatenate_videos.py

1672124306.393705 [gpodder.extensions] DEBUG: Found extension "enqueue_in_mediaplayer" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\enqueue_in_mediaplayer.py

1672124306.398717 [gpodder.extensions] DEBUG: Found extension "episode_website_context_menu" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\episode_website_context_menu.py

1672124306.403720 [gpodder.extensions] DEBUG: Found extension "filter" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\filter.py

1672124306.403720 [gpodder.extensions] DEBUG: Found extension "gtk_statusicon" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\gtk_statusicon.py

1672124306.408751 [gpodder.extensions] DEBUG: Found extension "minimize_on_start" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\minimize_on_start.py

1672124306.408751 [gpodder.extensions] DEBUG: Found extension "mpris-listener" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\mpris-listener.py

1672124306.418330 [gpodder.extensions] DEBUG: Found extension "normalize_audio" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\normalize_audio.py

1672124306.418330 [gpodder.extensions] DEBUG: Found extension "notification" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\notification.py

1672124306.418330 [gpodder.extensions] DEBUG: Found extension "notification-win32" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\notification-win32.py

1672124306.428427 [gpodder.extensions] INFO: Module loaded: D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\notification-win32.py

1672124306.428427 [gpodder.extensions] DEBUG: Found extension "rename_download" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\rename_download.py

1672124306.438501 [gpodder.extensions] DEBUG: Found extension "rm_ogg_cover" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\rm_ogg_cover.py

1672124306.443521 [gpodder.extensions] DEBUG: Found extension "rockbox_convert2mp4" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\rockbox_convert2mp4.py

1672124306.448555 [gpodder.extensions] DEBUG: Found extension "rockbox_coverart" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\rockbox_coverart.py

1672124306.448555 [gpodder.extensions] DEBUG: Found extension "sonos" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\sonos.py

1672124306.448555 [gpodder.extensions] DEBUG: Found extension "tagging" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\tagging.py

1672124306.458769 [gpodder.extensions] DEBUG: Found extension "taskbar_progress" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\taskbar_progress.py

1672124306.458769 [gpodder.extensions] DEBUG: Found extension "ted_subtitles" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\ted_subtitles.py

1672124306.468571 [gpodder.extensions] DEBUG: Found extension "ubuntu_appindicator" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\ubuntu_appindicator.py

1672124306.473574 [gpodder.extensions] DEBUG: Found extension "ubuntu_unity" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\ubuntu_unity.py

1672124306.478620 [gpodder.extensions] DEBUG: Found extension "update_feeds_on_startup" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\update_feeds_on_startup.py

1672124306.478620 [gpodder.extensions] DEBUG: Found extension "video_converter" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\video_converter.py

1672124306.478620 [gpodder.extensions] DEBUG: Found extension "youtube-dl" in D:\gpodder-3.11.0-portable 933\data\share\gpodder\extensions\youtube-dl.py

1672124306.488550 [gpodder.registry] DEBUG: Registering feed_handler resolver: <bound method SoundcloudFeed.fetch_channel of <class 'gpodder.plugins.soundcloud.SoundcloudFeed'>>

1672124306.488550 [gpodder.registry] DEBUG: Registering feed_handler resolver: <bound method SoundcloudFeed.fetch_channel of <class 'gpodder.plugins.soundcloud.SoundcloudFavFeed'>>

 

gPodder 3.11.0 (2022-07-30) - http://gpodder.org/

© 2005-2022 The gPodder Team

License: GNU General Public License, version 3 or later

 

Entering interactive shell. Type 'help' for help.

Press Ctrl+D (EOF) or type 'quit' to quit.

 

1672124306.498331 [gpodder.dbsqlite] INFO: Loading podcasts

1672124306.513751 [gpodder.dbsqlite] DEBUG: Database opened.

gpo> subscribe ***@***.***

1672124339.223653 [gpodder.youtube] DEBUG: Analyzing URL: https www.youtube.com ***@***.***

1672124339.233528 [urllib3.connectionpool] DEBUG: Starting new HTTPS connection (1): www.youtube.com:443

1672124339.424065 [urllib3.connectionpool] DEBUG: https://www.youtube.com:443 "GET ***@***.*** HTTP/1.1" 302 0

1672124339.424065 [urllib3.connectionpool] DEBUG: Starting new HTTPS connection (1): consent.youtube.com:443

1672124339.603948 [urllib3.connectionpool] DEBUG: https://consent.youtube.com:443 "GET /m?continue=https%3A%2F%2Fwww.youtube.com%2F%40MiraDoodles%3Fcbrd%3D1&gl=GB&m=0&pc=yt&hl=ru&src=1 HTTP/1.1" 302 0

1672124339.684044 [urllib3.connectionpool] DEBUG: https://consent.youtube.com:443 "GET ***@***.***?cbrd%3D1&gl=GB&hl=ru&pc=yt&src=1 HTTP/1.1" 200 None

1672124339.684044 [gpodder.youtube] WARNING: Could not retrieve youtube channel id.

Traceback (most recent call last):

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder\youtube.py", line 448, in get_channel_id_url

    xml_data = xml.etree.ElementTree.parse(raw_xml_data)

  File "C:/msys64/home/appveyor/_gpodder_build_root/mingw32/lib/python3.10/xml/etree/ElementTree.py", line 1222, in parse

  File "C:/msys64/home/appveyor/_gpodder_build_root/mingw32/lib/python3.10/xml/etree/ElementTree.py", line 580, in parse

xml.etree.ElementTree.ParseError: mismatched tag: line 239, column 200

1672124339.693931 [gpodder.log] ERROR: Uncaught exception: Traceback (most recent call last):

  File "<string>", line 1, in <module>

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 1295, in main

    cli._shell()

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 1154, in _shell

    self._parse(args)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 1264, in _parse

    return self._checkargs(func, command_line)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 1184, in _checkargs

    return func(*command_line)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 368, in subscribe

    existing = self.get_podcast(url, check_only=True)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 329, in get_podcast

    url = youtube.parse_youtube_url(url)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder\youtube.py", line 635, in parse_youtube_url

    new_url = get_channel_id_url(url)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder\youtube.py", line 463, in get_channel_id_url

    raise Exception('Could not retrieve youtube channel id.')

Exception: Could not retrieve youtube channel id.

 

Traceback (most recent call last):

  File "<string>", line 1, in <module>

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 1295, in main

    cli._shell()

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 1154, in _shell

    self._parse(args)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 1264, in _parse

    return self._checkargs(func, command_line)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 1184, in _checkargs

    return func(*command_line)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 368, in subscribe

    existing = self.get_podcast(url, check_only=True)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder_launch\gpo.py", line 329, in get_podcast

    url = youtube.parse_youtube_url(url)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder\youtube.py", line 635, in parse_youtube_url

    new_url = get_channel_id_url(url)

  File "D:\gpodder-3.11.0-portable 933\data\lib\python3.10\site-packages\gpodder\youtube.py", line 463, in get_channel_id_url

    raise Exception('Could not retrieve youtube channel id.')

Exception: Could not retrieve youtube channel id.

 

D:\gpodder-3.11.0-portable 933\data\bin>

 

Von: auouymous ***@***.***> 
Gesendet: Dienstag, 27. Dezember 2022 07:48
An: gpodder/gpodder ***@***.***>
Cc: computerfreak105 ***@***.***>; Author ***@***.***>
Betreff: Re: [gpodder/gpodder] GPO CLI crashes when entering youtube channel which doesn't contain channel ID (Issue #1448)

 

It has worked fine in both cli and gtk versions since the new URLs were introduced. Can you run gpo -v, subscribe to the URL, and then post the log file here? This will spam the console with a lot of log messages.

**Comment by auouymous** [comment]
Ah, you are in Europe, and the GDPR consent redirection is crashing it. That is a problem I'm unable to fix as I am not in Europe.

**Comment by computerfreak105** [comment]
Pity, thanks for trying anyway.
Is there something I could help with? 
I could let you work on my system over TeamViewer, if this would circumvent the location issue and if no one else can do it at the moment.

**Comment by auouymous** [comment]
Thanks for the offer, but you shouldn't give strangers access to your system. I also don't have TeamViewer and have no idea how to code on Windows.

**Comment by computerfreak105** [comment]
Ok, thanks, no problem. I wonder why this is dependent on the location?

**Comment by auouymous** [comment]
Because GDPR is only a law in Europe, so Google only shows the consent form to those accessing youtube from there.

**Comment by computerfreak105** [comment]
I found a workaround for those in Europe who are desperate enough: I used a VPN (Adguard) and set the location to united states new york. And then I could import a youtube url with @ without problems. I was curious what would happen in the XML file when I exported the subscriptions. turns out that GPO just resolves the channel names to the long traditional channel ID string we know. Which means if anyone would want to unsubscribe from the command line it's not possible to use the new channel names url. The channel ID string url is needed to unsubscribe.
I certainly can work with that for now.

Hope this helps.
